test/github-issues/6168/issue-6168.ts

```diff
@@ -0,0 +1,115 @@
+import {Table} from "../../../src/schema-builder/table/Table";
+import { QueryRunner, createConnection } from "../../../src";
+import { MysqlConnectionOptions } from "../../../src/driver/mysql/MysqlConnectionOptions";
+import { expect } from "chai";
+
+const questionName = "question";
+const categoryName = "category";
+
+const createDB = async (queryRunner: QueryRunner, dbName: string) => {
+    await queryRunner.createDatabase(dbName, true);
+    const questionTableName = `${dbName}.${questionName}`;
+    const categoryTableName = `${dbName}.${categoryName}`;
+
+    await queryRunner.createTable(new Table({
+        name: questionTableName,
+        columns: [
+            {
+                name: "id",
+                type: "int",
+                isPrimary: true,
+                isGenerated: true,
+                generationStrategy: "increment"
+            },
+            {
+                name: "name",
+                type: "varchar",
+            }
+        ],
+    }), true);
+
+    await queryRunner.createTable(new Table({
+        name: categoryTableName,
+        columns: [
+            {
+                name: "id",
+                type: "int",
+                isPrimary: true,
+                isGenerated: true,
+                generationStrategy: "increment"
+            },
+            {
+                name: "questionId",
+                type: "int",
+            }
+        ],
+        foreignKeys: [
+            {
+                columnNames: ["questionId"],
+                referencedTableName: questionTableName,
+                referencedColumnNames: ["id"],
+                name: "FK_CATEGORY_QUESTION"
+            }
+        ]
+    }), true);
+};
+
+describe("github issues > #6168 fix multiple foreign keys with the same name in a mysql multi-tenanted DB", () => {
+
+    let connections: Connection[];
+    let testConnections = [] as Connection[];
+    let testQueryRunners = [] as QueryRunner[];
+    before(async () => {
+        connections = await createTestingConnections({
+            entities: [__dirname + "/entity/*{.js,.ts}"],
+            enabledDrivers: ["mysql"],
+            schemaCreate: false,
+            dropSchema: false,
+        });
+
+        if (connections.length > 0) {
+            const queryRunner = connections[0].createQueryRunner();
+            await createDB(queryRunner, "test1");
+            await createDB(queryRunner, "test2");
+            await queryRunner.release();
+        };
+    });
+
+    beforeEach(() => reloadTestingDatabases(connections));
+
+    after(async () => {
+        if (connections.length > 0) {
+            const queryRunner = connections[0].createQueryRunner();
+            await queryRunner.dropDatabase("test1");
+            await queryRunner.dropDatabase("test2");
+            await queryRunner.release();
+        };
+
+        await closeTestingConnections(connections);
+        await Promise.all(testQueryRunners.map(queryRunner => queryRunner.release()));
```

Comment:
You can create and release this in the test itself - as long as it's before the `expect`